src/main/scala/firrtl/Emitter.scala

```diff
@@ -779,9 +792,15 @@ class VerilogEmitter extends SeqTransform with Emitter {
         emit(Seq("`endif // RANDOMIZE"))
       }
 
-      for (clk_stream <- at_clock if clk_stream._2.nonEmpty) {
-        emit(Seq(tab, "always @(posedge ", clk_stream._1, ") begin"))
-        for (x <- clk_stream._2) emit(Seq(tab, tab, x))
+      for ((clk, content) <- noResetAlwaysBlocks if content.nonEmpty) {
+        emit(Seq(tab, "always @(posedge ", clk, ") begin"))
+        for (line <- content) emit(Seq(tab, tab, line))
+        emit(Seq(tab, "end"))
+      }
+
+      for ((clk, reset, content) <- asyncResetAlwaysBlocks if content.nonEmpty) {
+        emit(Seq(tab, "always @(posedge ", clk, " or posedge ", reset, ") begin"))
```

Comment:
I think this then leaves the door open to eventually negedge triggered things (#695).

Reply:
Yeah, I think this would support that with minor modifications, I think the bigger issues involve how (if at all) negedge involves additionally checks and how (if at all) FIRRTL should reason about flops on the same clock but different edges should be handled.